trunk/SUAVE/Components/Energy/Networks/Ramjet.py

```diff
@@ -0,0 +1,243 @@
+
+# suave imports
+import SUAVE
+
+from SUAVE.Core import Data, Units
+from SUAVE.Components.Propulsors.Propulsor import Propulsor
+
+# ----------------------------------------------------------------------
+#  Ramjet Network
+# ----------------------------------------------------------------------
+
+## @ingroup Components-Energy-Networks
+class Ramjet(Propulsor):
+    """ This is a ramjet for supersonic flight.
+
+        Assumptions:
+        None
+
+        Source:
+        Most of the componentes come from this book:
+        https://web.stanford.edu/~cantwell/AA283_Course_Material/AA283_Course_Notes/
+    """
+
+    def __defaults__(self):
+	""" This sets the default values for the network to function.
+
+            Assumptions:
+            None
+
+            Source:
+            N/A
+
+            Inputs:
+            None
+
+            Outputs:
+            None
+
+            Properties Used:
+            N/A
+        """
+
+        #setting the default values
+        self.tag = 'Ramjet'
+        self.number_of_engines = 1.0
+        self.nacelle_diameter  = 1.0
+        self.engine_length     = 1.0
+
+    _component_root_map = None
+
+
+    def evaluate_thrust(self,state):
+	""" Calculate thrust given the current state of the vehicle
+
+		Assumptions:
+		None
+
+		Source:
+		N/A
+
+		Inputs:
+		state [state()]
+
+		Outputs:
+		results.thrust_force_vector                   [newtons]
+		results.vehicle_mass_rate                     [kg/s]
+		conditions.propulsion.acoustic_outputs:
+		    core:
+			exit_static_temperature                  [K]
+			exit_static_pressure                     [K]
+			exit_stagnation_temperature              [K]
+			exit_stagnation_pressure                 [Pa]
+			exit_velocity                            [m/s]
+		    fan:
+			exit_static_temperature                  [K]
+			exit_static_pressure                     [K]
+			exit_stagnation_temperature              [K]
+			exit_stagnation_pressure                 [Pa]
+			exit_velocity                            [m/s]
+
+		Properties Used:
+		Defaulted values
+	    """
+
+        #Unpack
+        conditions = state.conditions
+
+        ram                       = self.ram
+        inlet_nozzle              = self.inlet_nozzle
+        combustor                 = self.combustor
+        core_nozzle               = self.core_nozzle
+        thrust                    = self.thrust
+        number_of_engines         = self.number_of_engines
+
+        #Creating the network by manually linking the different components
+
+        #set the working fluid to determine the fluid properties
+        ram.inputs.working_fluid                               = self.working_fluid
+
+        #Flow through the ram , this computes the necessary flow quantities and stores it into conditions
+        ram(conditions)
+
+        #link inlet nozzle to ram
+        inlet_nozzle.inputs.stagnation_temperature             = ram.outputs.stagnation_temperature
```

Comment:
You don't need to connect every input to every output actually, just component output to the next component input:

for example:
inlet_nozzle.inputs =  ram.outputs